launcher/MMCZip.cpp

```diff
@@ -332,9 +333,20 @@ std::optional<QStringList> extractSubDir(QuaZip* zip, const QString& subdir, con
         }
 
         extracted.append(target_file_path);
-        QFile::setPermissions(target_file_path,
-                              QFileDevice::Permission::ReadUser | QFileDevice::Permission::WriteUser | QFileDevice::Permission::ExeUser);
-
+        auto fileInfo = QFileInfo(target_file_path);
+        if (fileInfo.isFile()) {
+            auto permissions = fileInfo.permissions();
+            auto maxPermisions = QFileDevice::Permission::ReadUser | QFileDevice::Permission::WriteUser | QFileDevice::Permission::ExeUser |
+                                 QFileDevice::Permission::ReadGroup | QFileDevice::Permission::ReadOther;
+            auto minPermisions = QFileDevice::Permission::ReadUser | QFileDevice::Permission::WriteUser;
+
+            auto newPermisions = (permissions & maxPermisions) | minPermisions;
+            if (newPermisions != permissions) {
+                if (!QFile::setPermissions(target_file_path, permissions)) {
```

Comment:
Unless I'm mistaken, this seems to not be changing the file's permissions, as it is setting to what they were to begin with